From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: IBM Codestream linux-2.6.16 october2005, patch 03-01
Patch-Mainline: 2.6.18-rc1

  Description: kernel: too many IPD machine checks allowed.
  Symptom:     CPU check stop in case of multiple IPD machine checks.
  Problem:     MAX_IPD_TIME is by a factor of ten too small. This means that ten
               times more instruction processing damage machine checks would be
               allowed in the intended time frame. This could result in a cpu
               check stop of a physical cpu.
  Solution:    Fix value of MAX_IPD_TIME.
  Problem-ID:  -
  Archs:       s390-31, s390-64

Acked-by: Hannes Reinecke <hare@suse.de>

--- linux-2.5/drivers/s390/s390mach.c	11 Apr 2006 10:41:43 -0000	1.35.2.1
+++ linux-2.5/drivers/s390/s390mach.c	29 Apr 2006 09:46:36 -0000	1.35.2.2
@@ -13,6 +13,7 @@
 #include <linux/sched.h>
 #include <linux/errno.h>
 #include <linux/workqueue.h>
+#include <linux/time.h>
 
 #include <asm/lowcore.h>
 
@@ -363,7 +364,7 @@
 }
 
 #define MAX_IPD_COUNT	29
-#define MAX_IPD_TIME	(5 * 60 * 100 * 1000) /* 5 minutes */
+#define MAX_IPD_TIME	(5 * 60 * USEC_PER_SEC) /* 5 minutes */
 
 /*
  * machine check handler.

